### PR TITLE
refactor(spaces): updated state to reducer hooks

### PR DESCRIPTION
**Pull request by GoPro16**
Removes 3 `useState` calls for one reducer that can handle all the scenarios.

### CODE REVIEW

File: packages/spaces/src/Spaces.js [hunk 1]
```diff
@@ -56,6 +56,31 @@ export const sanitizeSpaces = spaces => {
...
+    loading: false,
+    error,
+  }),
+  LOADING: (state, { loading }) => ({
+    ...state,
+    loading: loading !== undefined ? loading : !state.loading,
+  }),
+};
+
+const spacesReducer = (state, action) => actions[action.type](state, action);
```
Comment (jordan-a-young) [inline packages/spaces/src/Spaces.js:82]: The only thing I can think of to be cognizant of is what will happen if `actions[action.type] === undefined`.

Comment (GoPro16) [reply packages/spaces/src/Spaces.js:82]: 🤷‍♂  In this case we are safe but yeah I think if it was an app built using the state reducer there would need to be an undefined check.

Comment (jordan-a-young) [reply packages/spaces/src/Spaces.js:82]: That's what I was thinking as well.

Comment (nylon22) [inline packages/spaces/src/Spaces.js:82]: would there be value in exporting this?

**Pull request merged**